src/session/content/message_row/document.rs

```diff
@@ -243,4 +251,37 @@ impl MessageDocument {
             }
         }
     }
+
+    fn try_load_thumbnail(&self, message: &Message) {
+        if let MessageContent::MessageDocument(data) = message.content().0 {
+            let imp = self.imp();
+            if let Some(thumbnail) = data.document.thumbnail {
+                imp.file_thumbnail_picture.set_visible(true);
```

Comment:
I guess you should set it invisible somewhere else? Otherwise I don't see the point.

Reply:
I can set it invisible in else branch, if recycling widgets won't delete that picture. I don't know how messages are recycled

Reply:
Yes, you should delete it manually. Recycling is pretty simple, you just have to expect that `set_message()` can be called on with a different document message automatically.

Reply:
Check latest commit

Reply:
Ah, sorry, I missed that.